### PR TITLE
Add test to confirm Redis image version

### PR DESCRIPTION
**Pull request by sjmiller609**
**What this PR does / why we need it**:

**Special notes for your reviewer**:
@ashb for this comment: https://astronomerteam.slack.com/archives/CLRKTKHGU/p1580586195005300

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
n/a
- [ ] If a new distribution or Airflow verison is added, please add in .circleci/generate_circleci_config.py and run the script
- [ ] If a new distribution or Airflow verison is added, there is an 'onbuild' directory and Dockerfile in all base image directories
- [ ] If a new distribution is added, it is supported by all Airflow versions
- [ ] If a new Airflow version is added, it supports all distributions
- [ ] If changing an image, add applicable test in .circleci/test-airflow-image.py

### CODE REVIEW

**Comment by ashb** [comment]
Yes, but 3.4.1 is released that fixes the problem :)

**Comment by sjmiller609** [comment]
> Yes, but 3.4.1 is released that fixes the problem :)

@ashb do we want to check >= 3.4.1?

**Comment by ashb** [comment]
Yes, sounds good

I'm on mobile so can't check failures easily - are they falling this new check?

**Comment by sjmiller609** [comment]
looks like an unexpected error with CI "no space left on device", i'll just re-run

File: .circleci/test-airflow-image.py [hunk 1]
```diff
@@ -54,6 +54,17 @@ def test_elasticsearch_version(webserver):
         "elasticsearch module must be version 5.5.3 or greater"
 
 
+def test_redis_version(webserver):
+    """ Astronomer requires a version of redis greater than 3.4.1
```
Comment (ashb) [inline .circleci/test-airflow-image.py]:
```suggestion
    """ Astronomer requires a version of redis greater than or equal to 3.4.1
```

**Comment by sjmiller609** [comment]
Looks like some images are 3.3.11, so I made it check >= 3.4.1 or == 3.3.11

**Comment by ashb** [comment]
Ah yeah, ones that haven't been rebuilt since will fail I guess.

Do you think its worth adding a

```
redis!=3.4.0
```

to all the pip-constraint.txt files?

**Comment by sjmiller609** [comment]
Ok I switched up the test logic to test that it's NOT 3.4.0 and also I did the pip-constraints change as requested. rhel is failing because of a security issue.

**Comment by sjmiller609** [comment]
@schnie this one is low risk and fixes an active issue

**Pull request merged**